btree/btree.c

```diff
@@ -3152,13 +3153,33 @@ int64_t btree_create_tick(struct m0_sm_op *smop)
 int64_t btree_destroy_tick(struct m0_sm_op *smop)
 {
 	struct m0_btree_op 	*bop = M0_AMB(bop, smop, bo_op);
-	//ToDo: Implement complete destroy tick function.
+	struct m0_btree_oimpl	*oi = bop->bo_i;
 	switch(bop->bo_op.o_sm.sm_state)
 	{
 		case P_INIT:
-			return P_ACT;
+			M0_PRE(bop->bo_arbor != NULL);
+			M0_PRE(bop->bo_arbor->t_desc != NULL);
+			M0_PRE(node_invariant(bop->bo_arbor->t_desc->t_root));
+
+			/** The following pre-condition is currently a
+			 *  compulsion as the delete routine has not been
+			 *  implemented yet.
+			 *  Once it is implemented, this pre-condition can be
+			 *  modified to compulsorily remove the records and get
+			 *  the node count to 0.
+			 */
+			
+			M0_PRE(node_count(bop->bo_arbor->t_desc->t_root) == 0);
+			bop->bo_i = m0_alloc(sizeof *bop->bo_i);
+			oi = bop->bo_i;
 
-		case P_ACT:
+			tree_delete(&oi->i_nop, bop->bo_arbor->t_desc,
```

Comment:
I don't see `oi` getting used at any other place in this function. You can use `&bop->bo_i->i_nop` and avoid using extra `oi` variable.

Reply:
I used `oi` because of alignment.
I will update it.